src/dragonball/src/device_manager/mod.rs

```diff
@@ -188,6 +199,8 @@ pub struct DeviceOpContext {
     logger: slog::Logger,
     is_hotplug: bool,
 
+    #[cfg(feature = "hotplug")]
+    upcall_client: Option<Arc<UpcallClient<DevMgrService>>>,
     #[cfg(feature = "dbs-virtio-devices")]
```

Comment:
Why is virtio device an optional thing? Is there a use case we don't want devices at all? How do we handle vsock when virtio devices are disabled?

Reply:
There are currently no cases where no virtio devices are used. This is just to make it better when refactoring the code, without modifying a lot of virtio device code. Consider removing virtio features in the future.

The `hotplug` features depends on `dbs-upcall` and `virtio-vsock`.
The `virtio-vsock` and any virtio devices depend on `dbs-virtio-devices`.